### ISSUE TITLE
Support hover popup webview

### ISSUE THREAD

**Issue by samestep** [microsoft/vscode#175234]
This proposal is a more concrete version of #66493, which was closed as a duplicate of #3220. I am opening this new issue because I do not believe that the original was actually a duplicate: a hover popup is fundamentally different from an inline peek widget, as (unless I am mistaken) there can only be one hover popup visible at a time.

Specifically, I would like to add a new extension API that combines the [webview API](https://code.visualstudio.com/api/extension-guides/webview) with the [hover API](https://code.visualstudio.com/api/language-extensions/programmatic-language-features#show-hovers) (and possibly the [decoration API](https://vscode.rocks/decorations/)) to allow extensions to render custom interactive HTML content in popups, similar to the builtin color picker. At a high level, this should look the same as it currently does to provide a webview, except that the webview will be in a popup element floating above an existing editor, instead of in a separate pane.

I understand that the webview API docs mention that they are "resource heavy" and "should be used sparingly", which is why I believe the popup context is appropriate: since there can only be one at a time, this API should allow only a single webview to be created per extension, with that webview being shown on hover and hidden when the hover focus is lost, re-rendering with different content when hover focus reappears on a different item. Thus the resource costs of this proposal should be similar to those of existing webviews.

I really want this feature and I see that similar feature requests have been around for a while, so I plan to implement this myself and open a pull request, but I wanted to open this issue first to try to gauge the probability that such a pull request would eventually be accepted/merged, before starting work.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by VSCodeTriageBot** [comment]
:slightly_smiling_face: This feature request received a sufficient number of community upvotes and we moved it to our backlog. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!
